allennlp/nn/initializers.py

```diff
@@ -39,20 +72,27 @@ class InitializerApplicator:
     All parameters in the Module will be initialized.
     """
     def __init__(self,
-                 initializers: Dict[str, Callable[[torch.Tensor], None]] = None,
-                 default_initializer: Callable[[torch.Tensor], None] = torch.nn.init.normal) -> None:
+                 initializers: Dict[str, Initializer] = None,
+                 default_initializer: Initializer = Initializer.by_name('normal')(),
+                 exclude: Sequence[str] = None) -> None:
         """
         Parameters
         ----------
-        initializers : Dict[str, Callable[[torch.Tensor], None]], optional (default = {})
+        initializers : ``Dict[str, Callable[[torch.Tensor], None]]``, optional (default = {})
             A dictionary mapping parameter regexes to initializers to be applied to parameters
             matching the regex.
-        default_initializer : Callable[[torch.Tensor], None], optional, (default = torch.nn.init.normal)
+        default_initializer : ``Callable[[torch.Tensor], None]``, optional (default = torch.nn.init.normal)
             A default initializer, which will be used in the case that the Applicator encounters a parameter
             which does not match any of the regexes provided.
+        exclude : ``Sequence[str]``, optional (default=``[]``)
```

Comment:
Maybe we should raise an error in the case that any exclude regex also matches an explicitly provided regex. Something's probably gone wrong if that case occurs. Alternatively the exclude regexes should override everything - doesn't seem like having this conditional application really buys us anything.